app/models/thread.php

```diff
@@ -0,0 +1,114 @@
+<?php
+class Thread extends AppModel
+{
+    const MIN_STRING_LENGTH = 1;
+    const MAX_STRING_LENGTH = 80;
+
+    public $validation = array(
+       'title' => array(
+            'length' => array(
+               'validate_between', self::MIN_STRING_LENGTH, self::MAX_STRING_LENGTH,
+            ),
+        ),
+    );
+
+    public static function getAll($offset, $limit)
+    {
+        $threads = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM thread LIMIT {$offset}, {$limit}");
```

Comment:
Like this sir? Yes and still same error.
        $params = array( (int) $offset, (int) $limit);
        $rows = $db->rows("SELECT * FROM thread LIMIT ?, ?",$params);